src/include/Spinlock.h

```diff
@@ -16,88 +16,49 @@
 #ifndef CEPH_SPINLOCK_H
 #define CEPH_SPINLOCK_H
 
-#include "acconfig.h"
+#include <atomic>
 
-#include <pthread.h>
+namespace ceph {
 
-typedef struct {
-#ifdef HAVE_PTHREAD_SPINLOCK
-  pthread_spinlock_t lock;
-#else
-  pthread_mutex_t lock;
-#endif
-} ceph_spinlock_t;
-
-#ifdef HAVE_PTHREAD_SPINLOCK
-
-static inline int ceph_spin_init(ceph_spinlock_t *l)
+inline void simple_spin_lock(std::atomic_flag& lock)
 {
-  return pthread_spin_init(&l->lock, PTHREAD_PROCESS_PRIVATE);
+ while(lock.test_and_set(std::memory_order_acquire))
+  ;
 }
 
-static inline int ceph_spin_destroy(ceph_spinlock_t *l)
+inline void simple_spin_unlock(std::atomic_flag& lock)
 {
-  return pthread_spin_destroy(&l->lock);
+ lock.clear(std::memory_order_release);
```

Comment:
again, do we have a micro benchmark result to compare the performance with and without the change?